packages/@ember/-internals/metal/lib/cache.ts

```diff
@@ -0,0 +1,128 @@
+import { createCache as glimmerCreateCache, getValue, isConst } from '@glimmer/validator';
+
+/**
+  Ember uses caching based on trackable values to avoid updating large portions
+  of the application. This caching is exposed via a cache primitive that can be
+  used to cache a specific computation, so that it will not update and will
+  return the cached value until a tracked value used in its computation has
+  updated.
+
+  @module @glimmer/tracking/primitives/cache
+  @public
+*/
+
+/**
+  Receives a function, and returns a wrapped version of it that memoizes based on
+  _autotracking_. The function will only rerun whenever any tracked values used
+  within it have changed. Otherwise, it will return the previous value.
+
+  ```js
+  import { tracked } from '@glimmer/tracking';
+  import { createCache, getValue } from '@glimmer/tracking/primitives/cache';
+
+  class State {
+    @tracked value;
+  }
+
+  let state = new State();
+  let computeCount = 0;
+
+  let counter = createCache(() => {
+    // consume the state. Now, `counter` will
+    // only rerun if `state.value` changes.
+    state.value;
+
+    return ++computeCount;
+  });
+
+  getValue(counter); // 1
+
+  // returns the same value because no tracked state has changed
+  getValue(counter); // 1
+
+  state.value = 'foo';
+
+  // reruns because a tracked value used in the function has changed,
+  // incermenting the counter
+  getValue(counter); // 2
+  ```
+
+  @method createCache
+  @static
+  @for @glimmer/tracking/primitives/cache
+  @public
+*/
+export function createCache<T>(fn: () => T) {
+  return glimmerCreateCache(fn);
```

Comment:
Can you remind me why we can just export this directly? I know we talked about it, but I can’t recall if we had resolved

Reply:
ah, I remember we discussed this actually and it would be fine, any additional arguments are not document so it's fine, I'll update